### PR TITLE
Follow-up for Kokkos Sparse expval implementation

### PR DESCRIPTION
**Pull request by AmintorDusko**
**Context:**
This new Follow-up intends to implement more suggestions for the original PR.

**Description of the Change:**
- Cleaning code.
- Documentation improvement.
- Improvements to unit tests.

**Benefits:**
Improve the code quality.

**Possible Drawbacks:**
Not that I'm aware of.

**Related GitHub Issues:**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -98,13 +98,6 @@ test-cpp-kokkos:
 	cmake --build ./BuildTests --target runner
 	cmake --build ./BuildTests --target test
 
-.PHONY: benchmark
-benchmark:
-	cmake --build BuildBench --target clean || true
-	rm -rf ./BuildBench/CMakeCache.txt ./BuildBench/compiler_info.txt ./BuildBench/run_gate_benchmark.sh
-	cmake $(LIGHTNING_CPP_DIR) -BBuildBench -DBUILD_EXAMPLES=ON -DCMAKE_BUILD_TYPE=Release -DENABLE_AVX=ON
-	cmake --build ./BuildBench
-
```
Comment (mlxd) [inline Makefile:107]: Is this already gone from the master branch?

File: pennylane_lightning/src/util/Kokkos_Sparse.hpp [hunk 2]
```diff
@@ -155,12 +156,25 @@ void apply_Sparse_Matrix_Kokkos(
...
+ * @param vector_size   size of the vector.
+ * @param row_map_ptr   pointer to the row_map.
+ *                      The j element encodes the number of non-zeros above row
+ * j.
+ * @param row_map_size  number of elements in the row_map.
+ * @param entries_ptr   pointer to the column indices of the non-zero elements.
+ * @param values_ptr    non-zero elements.
+ * @param numNNZ        number of non-zero elements.
+ * @param result        result of the matrix vector multiplication.
+ * @param throw_exception if will true an exception.
```
Comment (mlxd) [inline pennylane_lightning/src/util/Kokkos_Sparse.hpp:177]:
```suggestion
 * @param throw_exception throws exception if true.
```

Though, I still think this as an argument does not make much sense.

File: Makefile (hunk 1 above)
Comment (maliasadi) [reply Makefile:107]: Yes! It's gone after adding the google-benchmark suite. There is now `gbenchmark`.

Comment (AmintorDusko) [reply Makefile:107]: It is still there.

**Review by maliasadi — COMMENTED** [review]
Thanks @AmintorDusko! Glad to see these updates.

File: Makefile (hunk 1 above)
Comment (AmintorDusko) [reply Makefile:107]: It was probably supposed to be gone, but for some reason, both are there.
We are fixing it now.

File: pennylane_lightning/src/util/Kokkos_Sparse.hpp [hunk 3]
```diff
@@ -155,12 +156,25 @@ void apply_Sparse_Matrix_Kokkos(
...
  *
- * @tparam fp_precision
- * @return std::vector<data_type<fp_precision>>
+ * @tparam fp_precision data float point precision.
+ * @tparam index_type integer type used as indices of the sparse matrix.
+ * @param vector_ptr    pointer to the vector.
+ * @param vector_size   size of the vector.
+ * @param row_map_ptr   pointer to the row_map.
+ *                      The j element encodes the number of non-zeros above row
+ * j.
```
[the remark this answers is not in the document]
Comment (AmintorDusko) [reply pennylane_lightning/src/util/Kokkos_Sparse.hpp:171]: clang-format changed it.

**Comment by AmintorDusko** [comment]
Hi @mlxd and @maliasadi. Please let me know if you have more suggestions or comments.
In case you have something to point out in other already merged PR, please let me know here too.

**Comment by mlxd** [comment]
Thanks @AmintorDusko 
I'm happy to approve, once @maliasadi  is.

File: pennylane_lightning/src/simulator/Measures.hpp [hunk 4]
```diff
@@ -162,6 +162,7 @@ class Measures {
     /**
```
Comment (maliasadi) [inline pennylane_lightning/src/simulator/Measures.hpp:162]: Could you please update the following lines,
- line 45: `@tparam SVType Statevector type to register`,
- line 197 ` @tparam op_type Operation type.`
- line 280 missing `op_type`

File: pennylane_lightning/src/util/Kokkos_Sparse.hpp [hunk 5]
```diff
@@ -79,8 +79,7 @@ using const_data_view_type =
  * @brief Create a Kokkos Sparse Matrix object with unmanaged views.
```
Comment (maliasadi) [inline pennylane_lightning/src/util/Kokkos_Sparse.hpp:79]: Could you please check the followings,
-  What does this "The j element encodes the number of non-zeros above row j" refer to at lines 82, 115, 168, 201?
- line 85 was that auto-formatted?

**Review by maliasadi — CHANGES_REQUESTED** [review]
Thank you @AmintorDusko for these follow-ups. Happy to approve after addressing a few minor updates.

File: pennylane_lightning/src/util/Kokkos_Sparse.hpp (hunk 5 above)
Comment (AmintorDusko) [reply pennylane_lightning/src/util/Kokkos_Sparse.hpp:79]: In order:

- Let me know if it is clear now.
- Yes. It was.

**Review by maliasadi — APPROVED** [review]

**Comment by chaeyeunpark** [comment]
Great job @AmintorDusko. I just would like to ask whether it is possible to add a parallel process option also in `Tidy (C++)`. The relevant workflow is defined in `workflows/format.yml`.

**Comment by AmintorDusko** [comment]
Hi @mlxd, @maliasadi and @chaeyeunpark. 
Thank you for your review!

**Pull request merged**